source/time_stepping/convection_time_step.cc

```diff
@@ -51,11 +57,36 @@ namespace aspect
             fe_values.reinit (cell);
             fe_values[this->introspection().extractors.velocities].get_function_values (this->get_solution(),
                                                                                         velocity_values);
+            in.reinit(fe_values, cell, this->introspection(), this->get_solution());
+            this->get_material_model().fill_additional_material_model_inputs(in, this->get_solution(), fe_values, this->introspection());
+            this->get_material_model().evaluate(in, out);
 
             double max_local_velocity = 0;
             for (unsigned int q=0; q<n_q_points; ++q)
-              max_local_velocity = std::max (max_local_velocity,
-                                             velocity_values[q].norm());
+              {
+                if (this->introspection().compositional_name_exists("porosity"))
+                  {
+                    const unsigned int porosity_idx = this->introspection().compositional_index_for_name("porosity");
+                    if (this->get_parameters().compositional_field_methods[porosity_idx] == Parameters<dim>::AdvectionFieldMethod::fem_darcy_field)
+                      {
+                        const Tensor<1,dim> gravity = this->get_gravity_model().gravity_vector(fe_values.quadrature_point(q));
+                        const double porosity = std::max(in.composition[q][porosity_idx], 0.0);
+                        const double solid_density = out.densities[q];
+                        // const double fluid_density = fluid_out->fluid_densities[q];
+                        // const double fluid_viscosity = fluid_out->fluid_viscosities[q];
+                        // const double permeability = fluid_out->permeabilities[q];
+                        const double fluid_density = 1000;
+                        const double fluid_viscosity = 10;
+                        const double permeability = 1e-6  * std::pow(porosity,3) * std::pow(1.0-porosity,2);
```

Comment:
`std::pow` is surprisingly slow. It doesnt matter here, but just to get used to it, if you need to take something to the power of an integer, a much faster version is:
```
Utilities::fixed_power<2,double>(1.0-porosity);
Utilities::fixed_power<3,double>(porosity);
```